include/picongpu/fields/laserProfiles/Wavepacket.hpp

```diff
@@ -0,0 +1,267 @@
+            // transform coordinate system to center of global simulation as origin [cells]
+            offsetToTotalDomain = /* totalCellOffset + */ globalCellOffset - halfSimSize;
+
+            // @todo reset origin of direction of moving window
+            // offsetToTotalDomain.y() = 0
+
+            float_64 const runTime = DELTA_T * currentStep - Unitless::laserTimeShift;
+
+            elong = float3_X::create( 0.0 );
+            float_X envelope = float_X( Unitless::AMPLITUDE );
+
+            const float_64 tau = Unitless::PULSE_LENGTH * math::sqrt( 2.0 );
+
+            float_64 correctionFactor = 0.0;
+
+            if( runTime > Unitless::startDownramp )
+            {
+                // downramp = end
+                const float_64 exponent =
+                    ( ( runTime - Unitless::startDownramp )
+                     / Unitless::PULSE_LENGTH / math::sqrt( 2.0 ) );
+                envelope *= math::exp( -0.5 * exponent * exponent );
+                correctionFactor = ( runTime - Unitless::startDownramp ) / ( tau * tau * Unitless::w );
+            }
+            else if( runTime < Unitless::endUpramp )
+            {
+                // upramp = start
+                const float_64 exponent = ( ( runTime - Unitless::endUpramp ) / Unitless::PULSE_LENGTH / math::sqrt( 2.0 ) );
+                envelope *= math::exp( -0.5 * exponent * exponent );
+                correctionFactor = ( runTime - Unitless::endUpramp ) / ( tau * tau * Unitless::w );
+            }
+
+            phase += float_X( Unitless::w * runTime ) + Unitless::LASER_PHASE;
+
+            if( Unitless::Polarisation == Unitless::LINEAR_X )
+            {
+                elong.x() = float_X( envelope * ( math::sin( phase ) + correctionFactor * math::cos( phase ) ) );
+            }
+            else if( Unitless::Polarisation == Unitless::LINEAR_Z )
+            {
+                elong.z() = float_X( envelope * ( math::sin( phase ) + correctionFactor * math::cos( phase ) ) );
+            }
+            else if( Unitless::Polarisation == Unitless::CIRCULAR )
+            {
+                elong.x() = float_X( envelope / math::sqrt( 2.0 ) * ( math::sin( phase ) + correctionFactor * math::cos( phase ) ) );
+                elong.z() = float_X( envelope / math::sqrt( 2.0 ) * ( math::cos( phase ) + correctionFactor * math::sin( phase ) ) );
+            }
+        }
+
+        /** create device manipulator functor
+         *
+         * @tparam T_WorkerCfg pmacc::mappings::threads::WorkerCfg, configuration of the worker
+         * @tparam T_Acc alpaka accelerator type
+         *
+         * @param alpaka accelerator
+         * @param localSupercellOffset (in supercells, without guards) to the
+         *        origin of the local domain
+         * @param configuration of the worker
+         */
+        template<
+            typename T_WorkerCfg,
+            typename T_Acc
+        >
+        HDINLINE acc::Wavepacket< Unitless >
+        operator()(
```

Comment:
Please rebase this PR: Operator must be const since the interface change in #2600